infra/license/README.md

```diff
@@ -0,0 +1,71 @@
+# License Verification
+
+## Background
+
+Numerous kinds of Node packages are needed to build and use ONE-vscode but some packages may have unacceptable license for both ONE and ONE-vscode.
+Therefore we should review whether licenses of each package can be used or not.
+However, if we check all of them manually, we may mistake or miss something.
+To reduce such risk and labor, automatic license checker would be good solution.
+
+## Process & Architecture
+
+### Flowchart
+
+![flowchart](flowchart.png)
+
+### Terminology
+
+- license-judgment.json
+  - Each of license names are got from the result of `license-checker`
+  - Each license is classfied to following categories
+    - `Allowed` : Allowed to use
+    - `Warning` : Manual review for each package is needed
+    - `Denied` : Not allowed to use
+- package-judgment.json
+  - Each judgment for a package consists of following information
+    ```json
+    "pkgname@version": {
+      "licenses": "License name",
+      "repository": "URL for package repository",
+      "ONE_permitted": "yes / conditional / no",
+      "ONE_caution": "(Optional field) Cautions when package is used"
+    },
+    ```
+    - `ONE_permitted` : Whether the package is allowed in ONE-vscode
+      - `yes` : Allowed to use
+      - `conditional` : Allowed to use under specific conditions. Needed to be checked before release.
+      - `no` : Not allowed to use
+    - `ONE_caution` : Comment about why `ONE_permitted` is `conditional` or `no`.
+
+### Guideline
+
+- Adding a license to `license-judgment.json`
+  - Add license name without any modification
+    - Do not remove parenthesis, `OR`, `AND` and so on
+- Adding a package to `package-judgment.json`
+  - Only add a package which cannot be judge only with `license-judgment.json`.
+  - If a package can be judged only with license name, do not add here. Unless, we shall add thousands of package judgments here.
+
+### Design Note
+
+- This flow is run on Ubuntu, Windows and MacOS
+  - Even same package is installed, dependent packages could be different
+- Why `package-judgment.json` is needed even we have `license-judgment.json`?
+  - If we only have `license-judgment.json`, following problems would be occurred
+    - If license is `UNKNOWN`, how can we know the license?
+    - If license is classified as `Warning`, can we judge only with license name whether it can be used or not? (e.g. `(MIT OR CC0-1.0)`)
+    - If package cannot be used but has `Allowed` license, how can we forbid the package?
+    - If package can be used but has `Denied` license, how can we permit the package?
```

Comment:
Just out of curiosity, what are the actual examples of this case?

Reply:
I couldn't find actual example for this :(
I just added this for contrast of `If package cannot be used but has Allowed license`.
Would it be better to remove this sentence and then add later if real example is occurred? :)

Reply:
It doesn't seem like such an important issue. It's just a question based on logical reasoning that it is unlikely that an allowable exception exists when use is prohibited at the license level, considering which of the license and the exceptions takes precedence and is more comprehensive.